pdi-dataservice-client/src/main/java/org/pentaho/di/core/jdbc/ThinUtil.java

```diff
@@ -655,4 +655,38 @@ public static String resolveFieldName( String field, RowMetaInterface serviceFie
     }
     return Optional.ofNullable( valueMeta );
   }
+
+  /**
+  * Remove first and last quote if string contains double quote
+  * */
+  public static String unQuote( String str ) {
+    StringBuilder builder = new StringBuilder( str );
+    if ( builder.indexOf( "\"\"" ) > 0 &&  builder.length() > 0 && builder.charAt( 0 ) == '"'
```

Comment:
Please replace this with:
    if ( str.startsWith("\"") && str.endsWith("\"")) {
      return str.substring(1, str.length() - 1 );
    } else {
      return str;
    }

Performance will be much better than using indexof and creating an intermediate object.